QUALITY_DECLARATION.md

```diff
@@ -1,132 +1,194 @@
 
 Additionally, there are generated header files which are installed and therefore part of the public API.
 The source templates for these generated headers are in the `resource` folder.
 
-## Change Control Process
+### API Stability Policy [1.iv]
 
-TODO fix link
+`rcutils` will not break public API within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
 
-`rcutils` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change_control_process).
+### ABI Stability Policy [1.v]
 
-This includes:
+`rcutils` contains C code and therefore must be concerned with ABI stability, and will maintain ABI stability within a ROS distribution.
 
-- all changes occur through a pull request
-- all pull request have two peer reviews
-- all pull request must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
-- all pull request must resolve related documentation changes before merging
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
 
-## Documentation
+`rcutils` will not break API nor ABI within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
 
-### Feature Documentation
+## Change Control Process [2]
 
-TODO fix link
+`rcutils` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process).
 
-`rcutils` has a [feature list](TODO) and each item in the list links to the corresponding feature documentation.
-There is documentation for all of the features, and new features require documentation before being added.
+### Change Requests [2.i]
+All changes will occur through a pull request, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
 
-### Public API Documentation
+### Contributor Origin [2.ii]
+All pull requests must have confirmation of contributor origin. This repository uses DCO.
 
-TODO fix link
+### Peer Review Policy [2.iii]
+All pull request will be peer-reviewed, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
 
-`rcutils` has embedded API documentation and it is generated using doxygen and is [hosted](TODO) along side the feature documentation.
-There is documentation for all of the public API, and new additions to the public API require documentation before being added.
+## Documentation [3]
 
-### License
+### Feature Documentation [3.i]
 
-The license for `rcutils` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the `LICENSE` file.
+`rcutils` does not have a documented feature list. Soon to be added.
```

Comment:
I think in the main page of the API live the feature list http://docs.ros2.org/eloquent/api/rcutils/index.html

Reply:
I think the correct link for the foxy release would be something like "[http://docs.ros2.org/foxy/api/rcutils/index.html](http://docs.ros2.org/foxy/api/rcutils/index.html)". However, that points to an outdated API. A quick check shows it is missing the function `rcutils_strerror`, which appears in the header files but not in that document